src/parseTools.js

```diff
@@ -1477,6 +1477,34 @@ function makeStaticString(string) {
   return '(stringToUTF8("' + string + '", ' + ptr + ', ' + len + '), ' + ptr + ')';
 }
 
+// Generates access to module exports variable in pthreads worker.js
+function makeAsmExportAccessInPthread(variable) {
+  if (MODULARIZE) {
+    return "Module['" + variable + "']" // 'Module' is defined in worker.js local scope, so not EXPORT_NAME in this case.
```

Comment:
hmm, `{{{ EXPORT_NAME }}}` is the name of the function that one needs to call to instantiate the compiled JS module - I should not be assigning anything over it?

`src/worker.js` currently does

```js
Module = {{{ EXPORT_NAME }}}(Module);
```
i.e. it uses `Module` as both the import object name, and also as export object name, like non-MODULARIZE-non-pthreads builds do.

Reply:
> hmm, {{{ EXPORT_NAME }}} is the name of the function that one needs to call to instantiate the compiled JS module - I should not be assigning anything over it?

True, yes - but I thought we were inside that function, so creating an alias might have simplified things? Did I get that wrong?

That is, what I meant to suggest is that for internal uses, we can do

```
function ExportName(Module) {
  var ExportName = Module;
  [.. internal uses here can use either Module or ExportName ..]
}
```
I think we do something similar in the general case, so that ExportName can be used internally.

Reply:
The worker global scope in `src/worker.js` is not inside the function scope, but it is in the worker's global scope outside the function, so it will either need to assign variables to global scope to import to the main JS code, and read variables off of global scope or Module, if not building with MODULARIZE. If building with MODULARIZE, imports need to go in an imports object, and exports need to be read off the return value of invoking the module function. I'm not sure how applicable the code snippet you write would be here?

Reply:
Ok, so if I understand you now, src/worker.js adds code to the global scope in both modularize and non-modularize mode?

Reply:
Yeah, the worker does add code to global scope in both cases. From inside the JS local scope in the module in MODULARIZE mode, it reads variables in the global scope. Strictly speaking this is not polluting, since the global scope of a pthread-hosting worker is not "public", i.e. there's no main .html file or external JS code that would potentially get injected to it. But perhaps in the future it may make sense to either formalize this, or move items to the imports object. I don't know yet.